### PR TITLE
fixed: Unable to upload annotations #4459

### LINKED ISSUE #4459: Unable to upload annotations
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour

- Annotations should be created through file upload.
### Current Behaviour

- After applying #4327, 403 Forbidden status code is returned when uploading a job annotation file.
- at the result of checking the opa log, the scope value of the input data is null.
- task annotations upload also same
- By #4327, when the "upload annotations" function is executed, the http request method is changed from PUT to POST, but the cause is that the new key:value definition for POST does not exist in get_scopes() of JobPermission.

image

### Possible Solution

- Add the following two key:value pairs to get_scopes() in JobPermission and TaskPermission class.
> ('annotations', 'POST'): 'import:annotations'
 > ('append_annotations_chunk', 'PATCH'): 'update:annotations',
### Steps to Reproduce (for bugs)

1. create some labeled shapes
1. save and export annotations (w/ task owner account)
1. remove annotations and save
1. upload annotations

### Your Environment

- Git hash commit (`git log -1`): 7ef553fd
- Docker version `docker version` (e.g. Docker 17.0.05):
- Are you using Docker Swarm or Kubernetes?
- Operating System and version (e.g. Linux, Windows, MacOS): ubuntu 18.04
- Code example or link to GitHub repo or gist to reproduce problem:
- Other diagnostic information / logs:
    <details>
    <summary>Logs from `cvat` container</summary>
    </details>

### Next steps
You may [join our Gitter](https://gitter.im/opencv-cvat/public) channel for community support.

### PR DESCRIPTION
**Pull request by k1won**
### Motivation and context

Resolves #4459
### How has this been tested?

Manual test
### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by nmanovic** [comment]
@klakhov , could you please look?

File: cvat/apps/iam/permissions.py [hunk 1]
```diff
@@ -645,6 +645,8 @@ def get_scopes(request, view, obj):
             ('annotations', 'PATCH'): 'update:annotations',
             ('annotations', 'DELETE'): 'delete:annotations',
             ('annotations', 'PUT'): 'update:annotations',
+            ('annotations', 'POST'): 'import:annotations',
+            ('append_annotations_chunk', 'PATCH'): 'update:annotations',
```
Comment (klakhov) [inline cvat/apps/iam/permissions.py:649]: I suppose we need to add HEAD method here:
```('append_annotations_chunk', 'HEAD'): 'update:annotations'```
It's also a part of TUS protocol, even though it does not generate an error in our case

File: cvat/apps/iam/permissions.py [hunk 2]
```diff
@@ -790,6 +792,8 @@ def get_scopes(request, view, obj):
...
```
Comment (klakhov) [inline cvat/apps/iam/permissions.py:796]: The same thing with HEAD here

**Review by klakhov — COMMENTED** [review]
Thanks for contribution! I've tried this patch and it fixes annotations upload to Task and Job.

But while investigating permissions issue, I've found one more problem with Task data upload: upload action was for Task data was renamed from `append_tus_chunk` to `append_data_chunk`. So could you change this permission too (lines 654-655)?

File: cvat/apps/iam/permissions.py (hunk 2 above)
Comment (k1won) [reply cvat/apps/iam/permissions.py:797]: Done. Thanks for your comment.

**Review by klakhov — APPROVED** [review]

**Review by nmanovic — APPROVED** [review]

**Comment by nmanovic** [comment]
@k1won , great contribution once again!

**Pull request merged**